blockmanager.go

```diff
@@ -493,6 +625,89 @@ func (b *blockManager) getAllCFHeaders(checkpoints []*chainhash.Hash,
 	)
 }
 
+// writeCFHeadersMsg writes a cfheaders message to the specified store. It
+// assumes that everything is being written in order. The hints are required
+// to store the correct block heights for the filters.
+func (b *blockManager) writeCFHeadersMsg(msg *wire.MsgCFHeaders,
+	store *headerfs.FilterHeaderStore) error {
+
+	// Check that the PrevFilterHeader is the same as the last stored so
+	// we can prevent misalignment.
+	tip, _, err := store.ChainTip()
+	if err != nil {
+		return err
+	}
+	if *tip != msg.PrevFilterHeader {
+		return fmt.Errorf("attempt to write cfheaders out of order")
+	}
+
+	// Cycle through the headers and create a batch to be written.
+	lastHeader := msg.PrevFilterHeader
+	headerBatch := make([]headerfs.FilterHeader, 0, wire.CFCheckptInterval)
+
+	for _, hash := range msg.FilterHashes {
+		lastHeader = chainhash.DoubleHashH(append(hash[:],
+			lastHeader[:]...))
+
+		fHeader := headerfs.FilterHeader{
+			FilterHash: lastHeader,
+		}
+		headerBatch = append(headerBatch, fHeader)
+	}
+
+	// Add the block hashes by walking backwards through the headers.
+	// Also, cache the block headers for notification dispatch.
+	processedHeaders := make([]*wire.BlockHeader, len(headerBatch))
+	curHeader, height, err := b.server.BlockHeaders.FetchHeader(
+		&msg.StopHash)
```

Comment:
Ignoring the error here.

Reply:
Not a blocker right now, but ideally we also _batch fetch_ these headers just to reduce the read I/O here. They can really all be fetched in one sweep. So at the very least, can you leave a TODO here? Thanks!